### ISSUE TITLE
Poetry appears to attempt to connect to pypi.org though I have it disabled

### ISSUE THREAD

**Issue by smprather** [python-poetry/poetry#7472]
- **Poetry version**: Poetry (version 1.3.2)
- **Python version**: Poetry (version 1.3.2)
- **OS version and name**: Poetry (version 1.3.2)
- **pyproject.toml**: https://gist.github.com/smprather/2a9e338226b53a9e3199e18b03ffcf50

- [ x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) stable Poetry version, installed using a recommended method.
- [ x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [ x] I have consulted the [FAQ](https://python-poetry.org/docs/faq/) and [blog](https://python-poetry.org/blog/) for any relevant entries or release notes.
- [ x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option) and have included the output below.

## Issue
I am operating on an internal network that doesn't have visibility of the external internet. We have an internal PyPI that we use. I am able to 'poetry add' many things (yaml, pytest, plotly, kaleido, and more). But when I try 'poetry add fire', it hangs for a while waiting on network timeouts and retries, then dies. The poetry add fire -vvv is in the pyproject.toml gist.

I tried manually running the 'pip install' spawned by Poetry (after activating the Poetry virtenv). I get the same hang. When I remove --isolated, it runs ok (I have ~/.config/pip/pip.conf pointed to our internal PyPI). Also, I hacked subprocess to print out the commands it is getting, and I see this one.

```python
['/home/epramyl/.cache/pypoetry/virtualenvs/redhawk-dvd-utils-e0EwSbsW-py3.9/bin/python', '/home/epramyl/.cache/pypoetry/virtualenvs/redhawk-dvd-utils-e0EwSbsW-py3.9/lib/python3.9/site-packages/pip/__pip-runner__.py', 'install', '--ignore-installed', '
--no-user', '--prefix', '/tmp/pip-build-env-vsjy__y3/overlay', '--no-warn-script-location', '--no-binary', ':none:', '--only-binary', ':none:', '-i', 'https://pypi.org/simple', '--', 'setuptools>=40.8.0', 'wheel']
```

Shouldn't Poetry be passing my configured source url as a --index-url argument to pip since it is using --isolated?

I would suspect user error (and still do), but I am able to add so many dependencies successfully. This is the first one that didn't work. When I create a fresh virt-env, activate, then pip install fire, no problem. It respects my pip.conf, and all is well.

Thanks!
--Myles

PS I have tried clearing the cache and recreating poetry.lock many times. Didn't help.

**Comment by neersighted** [comment]
Duplicate of #3249.
